src/CSharp/CodeCracker/Design/MakeMethodStaticAnalyzer.cs

```diff
@@ -67,8 +72,39 @@ private static void AnalyzeMethod(SyntaxNodeAnalysisContext context)
                 if (!dataFlowAnalysis.Succeeded) return;
                 if (dataFlowAnalysis.DataFlowsIn.Any(inSymbol => inSymbol.Name == "this")) return;
             }
+
+            if (IsTestMethod(method, methodSymbol)) return;
+
             var diagnostic = Diagnostic.Create(Rule, method.Identifier.GetLocation(), method.Identifier.ValueText);
             context.ReportDiagnostic(diagnostic);
         }
+
+        private static bool IsTestMethod(MethodDeclarationSyntax method, IMethodSymbol methodSymbol)
+        {
+            var result = false;
+
+            // Test if the method has any known test framework's attribute.
+            result = method.AttributeLists.HasAnyAttribute(AllTestFrameworksMethodAttributes.Value);
+
+            if (!result && methodSymbol.Name.Contains("Test"))
+            {
+                // Test if the containing class has any NUnit class attribute
+                result = methodSymbol.ContainingType.GetAttributes().Any(attribute => attribute.AttributeClass.Name == NUnitTestClassAttribute);
```

Comment:
This does not work, because the attribute name is actually reported as `TestFixtureAttribute` (it is a symbol).
Please add tests that consider this. You may have to actually declare the attribute classes on the tests to get this to work. It is easy to manually test, simply run and add NUnit as a reference, and create a test class with the `TestFixture` attribute with a simple method. Like this:

````csharp
using NUnit.Framework;
[TestFixture]
class Bar
{
    void Testa() { }
}
````